---
title: "Rhizospheric fungal/bacteria metagenome project"
format: html
editor: 
  markdown: 
    wrap: sentence
---

# Upload raw data files to server

The goal for week 2 is to upload fastq files to Raven server.
Note that this needs to be done from the local terminal, not the one on the server.

```{bash}
# move files from lab drive to Raven
scp "/Volumes/CRUZER16GB/F1B-KM40_R1_001.fastq.gz" \
rendavis@raven.fish.washington.edu:/home/shared/16TB_HDD_01/fish546/renee/
```

Ok that worked, now to upload the rest using rsync instead (better for handling multiple files, fewer chances of duplicates, conflicts etc).

```{bash}
# move remaining files from lab drive to Raven
rsync -av --progress \
    /Volumes/CRUZER16GB/*.fastq.gz \
    rendavis@raven.fish.washington.edu:/home/shared/16TB_HDD_01/fish546/renee/
```

# Quality control

I have been told that QC has been run on these, but we won't take that at face value.
For this we will use FastQC.

```{bash}
# fun FastQC on the fastq files
/home/shared/FastQC-0.12.1/fastqc -t 36 -o output /home/shared/16TB_HDD_01/fish546/renee/*.fastq.gz
```

```{bash}
Analysis complete for F2R-KM41_R1_001.fastq.gz
Analysis complete for F2R-KM41_R2_001.fastq.gz
Analysis complete for F1B-KM40_R1_001.fastq.gz
Analysis complete for F1B-KM40_R2_001.fastq.gz
```

.png files downloaded to my desktop.

# Check MultiQC

```{bash echo=TRUE}
eval "$(/opt/anaconda/anaconda3/bin/conda shell.bash hook)"
conda activate
which multiqc

cd output

multiqc .
```
See html report in output folder. 

# Next steps

These are R1/R2 (forward and reverse reads) and will have to be merged.
That will be next week.
I can attempt to follow [this guide](https://www.cureffi.org/2012/12/19/forward-and-reverse-reads-in-paired-end-sequencing/).
Also, need to create some subdirectories to better organize this as there will be some intermediary files.

Then, we will check to see if there is fungal and bacterial DNA.
