{
  "cells": [
    {
      "cell_type": "raw",
      "metadata": {},
      "source": [
        "---\n",
        "title: \"NifHunt: Soil metagenome N fixation analysis from BioBead greenhouse study\"\n",
        "author: Renee Davis\n",
        "format:\n",
        "  revealjs: \n",
        "    theme: simple\n",
        "    code-overflow: scroll\n",
        "execute:\n",
        "  eval: false\n",
        "  jupyter: false\n",
        "---"
      ],
      "id": "939659f4"
    },
    {
      "cell_type": "markdown",
      "metadata": {},
      "source": [
        "## Overview\n",
        "\n",
        "Nitrogen (N) fixation is a vital microbial process in soils. N-fixing microbes fox atmospheric N2 to a plant-usable form (NH3) via nitrogenase genes (*nif*H, *nif*D, *nif*K).\n",
        "\n",
        "I analyzed the metagenomes from 2 (of 48 total) soil samples that were used a greenhouse study prior to BioBead treatment.\n",
        "\n",
        "-   F1B - bulk field soil\n",
        "-   F2R - rhizosphere (root space) soil The goal is to identify the microbial communities in these 2 samples, analyze shared and unique taxa, and screen each for the 3 *nif* genes. Ultimately, I wanted to establish a pipeline for analyzing the rest of the samples for microbial ID and functional insights.\n",
        "\n",
        "## BioBead project\n",
        "\n",
        "## Challenges in soil biodiversity and inoculation\n",
        "\n",
        "## Methods\n",
        "\n",
        "-   Confirm data integrity with checksums\n",
        "-   Quality control: Fast QC and MultiQC\n",
        "-   Preprocessing: trimming via Trimmomatic, merging with PEAR\n",
        "-   Assembly via Megahit\n",
        "-   Phylogenetic tree construction via Megan\n",
        "-   Taxonomic identification via taxize\n",
        "-   Prodigal for gene prediction (protein coding genes)\n",
        "-   HMMER for nif gene screening\n",
        "\n",
        "## Trimming\n",
        "\n",
        "Trimming removes adapters and low-quality bases.\n",
        "\n",
        "\n",
        "```{bash}\n",
        "java -jar /home/shared/16TB_HDD_01/fish546/renee/Trimmomatic-0.39/trimmomatic-0.39.jar PE -threads 4\n",
        "  F1B-KM40_R1_001.fastq.gz F1B-KM40_R2_001.fastq.gz \\\n",
        "  F1B-KM40_trimmed_R1_paired.fastq.gz F1B-KM40_trimmed_R1_unpaired.fastq.gz \\\n",
        "  F1B-KM40_trimmed_R2_paired.fastq.gz F1B-KM40_trimmed_R2_unpaired.fastq.gz \\\n",
        "  ILLUMINACLIP:TruSeq3-PE.fa:2:30:10 \\\n",
        "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:20 MINLEN:50  #keeps bps equal or above 50 \n",
        "```\n",
        "\n",
        "\n",
        "This produces **unpaired** and **paired** file outputs. Paired reads are those which both forward and reverse survived trimming. These are used for downstream analysis like merging and assembly. Unpaired reads indicate where only one of the pair survived (the other was discarded due to low quality or short length).\n",
        "\n",
        "## Merging\n",
        "\n",
        "These are R1/R2 (forward and reverse reads) and will have to be merged. This is the last component of pre-processing as we work towards metagenome assembly.\n",
        "\n",
        "``` bash\n",
        "/home/shared/fastp-v0.24.0/fastp \\\n",
        "  -i F1B-KM40_trimmed_R1_paired.fastq.gz \\\n",
        "  -I F1B-KM40_trimmed_R2_paired.fastq.gz \\\n",
        "  --merge \\\n",
        "  --merged_out F1B-KM40_merged.fastq.gz \\ \n",
        "  \n",
        "/home/shared/fastp-v0.24.0/fastp \\\n",
        "  -i F2R-KM41_trimmed_R1_paired.fastq.gz \\\n",
        "  -I F2R-KM41_trimmed_R2_paired.fastq.gz \\\n",
        "  --merge \\\n",
        "  --merged_out F2R-KM41_merged.fastq.gz \n",
        "```\n",
        "\n",
        "## Assembly\n",
        "\n",
        "To assemble the metagenome files, MEGAHIT was used.\n",
        "\n",
        "``` {.bash code-line-numbers=\"2-4\"}\n",
        "./megahit \n",
        "  -r ../F1B-KM40_merged.fastq.gz  #specifying input file\n",
        "  -o megahit_F1B_KM40_out   #output directory\n",
        "  --min-contig-len 500  #over 500 bps\n",
        "  -t 8  #8 threads\n",
        "```\n",
        "\n",
        "Like other steps, this is done with both files.\n",
        "\n",
        "``` {.bash code-line-numbers=\"2-4\"}\n",
        "./megahit \\\n",
        "  -r ../F2R-KM41_merged.fastq.gz \\\n",
        "  -o megahit_F2R_KM41_out \\\n",
        "  --min-contig-len 500 \\\n",
        "  -t 8\n",
        "```\n",
        "\n",
        "## Phylogenetic tree construction via MEGAN\n",
        "\n",
        "Phylogenetic trees are constructed using MEGAN. This is the result for F2R, the rhizosphere soil. The code and results for F1B will be included in the next update.\n",
        "\n",
        "<embed src=\"https://gannet.fish.washington.edu/seashell/snaps/sr-blastx-meganized.pdf\" width=\"100%\" height=\"500px\" />\n",
        "\n",
        "## Rhizosphere soil top hits\n",
        "\n",
        "| Organism                      | Classification |\n",
        "|-------------------------------|----------------|\n",
        "| Acidobacteria bacterium       | Bacteria       |\n",
        "| Alphaproteobacteria bacterium | Bacteria       |\n",
        "| Betaproteobacteria bacterium  | Bacteria       |\n",
        "| Verrucomicrobia bacterium     | Bacteria       |\n",
        "| Actinobacteria bacterium      | Bacteria       |\n",
        "\n",
        ": Table 1: Rhizosphere soil top hits\n",
        "\n",
        "## Bulk soil top hits\n",
        "\n",
        "## Shared vs unique taxa\n",
        "\n",
        "## nif gene analysis\n",
        "\n",
        "## Prodigal to parse protein-coding genes\n",
        "\n",
        "## HMMER for nif gene screening\n",
        "\n",
        "## nif genes in metagenome samples\n",
        "\n",
        "## Conclusions\n",
        "\n",
        "## Next steps: scaling up"
      ],
      "id": "fa533bf6"
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3",
      "language": "python",
      "name": "python3"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}